src/HealthGPS/analysis_module.cpp

```diff
@@ -26,6 +26,46 @@ SimulationModuleType AnalysisModule::type() const noexcept {
     return SimulationModuleType::Analysis;
 }
 
+void AnalysisModule::initialise_map(RuntimeContext &context) {
+    // Set some arbitrary factors
+    auto factor1 = context.mapping().entries().at(0).key().to_string();
+    auto factor2 = context.mapping().entries().at(1).key().to_string();
+
+    // Get the minimum and maximum of the two factors in the population
+    auto min_max = std::minmax_element(context.population().cbegin(), context.population().cend(),
```

Comment:
While this is totally fine as it is, in C++20 you *can* write this as:

```suggestion
    auto min_max = std::ranges::minmax_element(context.population(),
```